src/main/java/org/jenkinsci/plugins/github/pullrequest/builders/GitHubPRStatusBuilder.java

```diff
@@ -52,41 +56,44 @@ public GitHubPRMessage getStatusMessage() {
     }
 
     @Override
-    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener)
-            throws InterruptedException, IOException {
-        GitHubPRTrigger trigger = build.getProject().getTrigger(GitHubPRTrigger.class);
+    public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull Launcher launcher,
+                        @Nonnull TaskListener listener) throws InterruptedException, IOException {
+        // No triggers in Run class, but we need it
+        final GitHubPRTrigger trigger = JobInfoHelpers.triggerFrom(run.getParent(), GitHubPRTrigger.class);
```

Comment:
for what it's worth, I'd like to see something like that method (and probably the whole class) in core somewhere to deal with the Run/AbstractProject pain.